packages/@lwc/template-compiler/src/shared/types.ts

```diff
@@ -142,6 +142,7 @@ export interface AbstractBaseElement extends BaseParentNode {
     listeners: EventListener[];
     directives: ElementDirective[];
     namespace?: string;
+    keyScope?: string;
 }
```

Comment:
I couldn't think of a better place to put this than on the `AbstractBaseElement`. It feels a bit weird, though.

Reply:
I'm sure we could, but I guess my mental model was that codegen should not be traversing up the parent tree, and that this made more sense to happen at parse time. @jmsjtu Do you have any thoughts on this?

Reply:
Fixed it so it happens only at codegen time. It's still a bit ugly IMO, so if you have suggestions, please let me know!